agent/consul/controller/queue_test.go

```diff
@@ -6,28 +6,30 @@ package controller
 import (
 	"sync/atomic"
 	"time"
+
+	"github.com/hashicorp/consul/agent/consul/controller/queue"
 )
 
-var _ WorkQueue = &countingWorkQueue{}
+var _ queue.WorkQueue[string] = &countingWorkQueue[string]{}
 
-type countingWorkQueue struct {
+type countingWorkQueue[T queue.ItemType] struct {
```

Comment:
I'm wondering if it makes more sense to move this into the new subpackage and then just expose it publicly for use in tests... I think if that ended up happening though it can probably wait for now and this is fine for the time being.